packages/hardhat-core/src/internal/artifacts.ts

```diff
@@ -40,9 +40,22 @@ import {
 
 const log = debug("hardhat:core:artifacts");
 
+interface Cache {
+  artifactPaths?: string[];
+  debugFilePaths?: string[];
+  buildInfoPaths?: string[];
+  artifactNameToArtifactPathCache: Map<string, string>;
+  artifactFQNToBuildInfoPathCache: Map<string, string>;
+}
+
 export class Artifacts implements IArtifacts {
   private _validArtifacts: Array<{ sourceName: string; artifacts: string[] }>;
 
+  private _cache?: Cache = {
```

Comment:
Let's add a comment saying that an undefined `_cache` signals a disabled cache.

Reply:
done